### DISCUSSION TITLE
Different permissions on WSL and Git Bash(MINGW64 shell)

### DISCUSSION THREAD

**Discussion by Anutrix** [microsoft/WSL#9285] [General]
### Version

Microsoft Windows [Version 10.0.22621.819]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.74.2-microsoft-standard-WSL2

### Distro Version

Kali GNU/Linux Rolling 2022.4

### Other Software

```
$ git version
git version 2.37.1.windows.1
```

### Repro Steps

1. See file permissions on WSL and Git Bash.

The behavior is inconsistent but happens often. I'll add better repro steps when I can.

### Expected Behavior

Both WSL and Git Bash on Windows should show same permissions. At least git should treat the changes same.

### Actual Behavior

Left is Git Bash and right is WSL.
![image](https://user-images.githubusercontent.com/11459028/205387835-0c40c39e-4d11-4cb0-bfcd-ba9b191d7775.png)

Note the file permissions for `starting_point_anutrix.ovpn` and also response of git status.
Both are different in both cases.

**Comment by elsaco** [comment]
@Anutrix it depends of your vantage point. When you list from inside a Linux shell the file ownership is applied per mount instructions:

`2373 2104 0:258 / /mnt/c rw,noatime - 9p drvfs rw,dirsync,aname=drvfs;path=C:\;uid=1000;gid=1000;symlinkroot=/mnt/,mmap,access=client,msize=262144,trans=virtio`

Notice that `C:\` is mounted with the default WSL UID/GID (1000). If you type `id` it will show current user's info:

`uid=1000(elsaco) gid=1000(elsaco) groups=1000(elsaco),492(dialout),1001(docker)`

So everything under `/mnt/c` will have UID/GID set to 1000 which in my case is my WSL username `elsaco` set during install.

---

Inside a Git Bash session it looks different:

`uid=197609(tito) gid=197609 groups=197609`

and `C:\` is mounted as:

`3 3 41634:16891 / /c binary,noacl,posix=0,user,noumount,auto - ntfs C: rw`


Use `stat` on the same file and you'll see the differences both shells display. In Linux UID/GID is very easy to see where is coming from (`/etc/passwd` and `/etc/group`) but in Git Bash I don't know where the shell gets the ownership info. Maybe it's random just to keep the shell happy.

**Comment by benhillis** [comment]
Wsl is not attempting to replicate git bash's environment.
